src/components/Compiler.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+class Compiler extends React.PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            dir: '',
+            text: ':/',
+            lastWord: '',
+            constTextLength: 2,
+            cursorVisible: true
+        };
+        this.cursorChar = '|';
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+        this.KEYCODE = {
+            backspace : 8,
+            enter: 13,
+            space: 32
+        };
+    }
+
+    changeCursorVisibility() {
+        this.setState({ cursorVisible: !this.state.cursorVisible });
```

Comment:
You can achieve this effect using pure css. It will improve performance, and ease your component out of some logic. There is plenty tutorials online on creating blinking cursor. Consider doing it this way.